cloud/amazon/sqs_queue.py

```diff
@@ -0,0 +1,221 @@
+            result['changed'] = True
+
+    except BotoServerError:
+        result['msg'] = 'Failed to create/update sqs queue due to error: ' + traceback.format_exc()
+        module.fail_json(**result)
+    else:
+        module.exit_json(**result)
+
+
+def update_sqs_queue(queue,
+                     check_mode=False,
+                     default_visibility_timeout=None,
+                     message_retention_period=None,
+                     maximum_message_size=None,
+                     delivery_delay=None,
+                     receive_message_wait_time=None):
+    changed = False
+
+    changed = set_queue_attribute(queue, 'VisibilityTimeout', default_visibility_timeout,
+                                  check_mode=check_mode) or changed
+    changed = set_queue_attribute(queue, 'MessageRetentionPeriod', message_retention_period,
+                                  check_mode=check_mode) or changed
+    changed = set_queue_attribute(queue, 'MaximumMessageSize', maximum_message_size,
+                                  check_mode=check_mode) or changed
+    changed = set_queue_attribute(queue, 'DelaySeconds', delivery_delay,
+                                  check_mode=check_mode) or changed
+    changed = set_queue_attribute(queue, 'ReceiveMessageWaitTimeSeconds', receive_message_wait_time,
+                                  check_mode=check_mode) or changed
+    return changed
+
+
+def set_queue_attribute(queue, attribute, value, check_mode=False):
+    if not value:
+        return False
+
+    existing_value = queue.get_attributes(attributes=attribute)[attribute]
+    if str(value) != existing_value:
+        if not check_mode:
+            queue.set_attribute(attribute, value)
+        return True
+
+    return False
+
+
+def delete_sqs_queue(connection, module):
+    queue_name = module.params.get('name')
+
+    result = dict(
+        region=module.params.get('region'),
+        name=queue_name,
+    )
+
+    try:
+        queue = connection.get_queue(queue_name)
+        if queue:
+            if not module.check_mode:
+                connection.delete_queue(queue)
+            result['changed'] = True
+
+        else:
+            result['changed'] = False
+
+    except BotoServerError:
+        result['msg'] = 'Failed to delete sqs queue due to error: ' + traceback.format_exc()
+        module.fail_json(**result)
+    else:
+        module.exit_json(**result)
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        state=dict(default='present', choices=['present', 'absent']),
+        name=dict(required=True, type='str'),
+        default_visibility_timeout=dict(type='int'),
+        message_retention_period=dict(type='int'),
+        maximum_message_size=dict(type='int'),
+        delivery_delay=dict(type='int'),
+        receive_message_wait_time=dict(type='int'),
+    ))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True)
+
```

Comment:
You need to add a handler for HAS_BOTO
E.g.
+    if not HAS_BOTO:
+        module.fail_json(msg='boto required for this module')